source/yats/include/yats/TaskConfigurator.h

```diff
@@ -0,0 +1,116 @@
+
+#pragma once
+
+#include <memory>
+#include <map>
+
+#include "Identifier.h"
+#include "InputConnector.h"
+#include "Nodecontainer.h"
+#include "OutputConnector.h"
+
+
+/**/
+class AbstractTaskConfigurator
+{
+public:
+	AbstractTaskConfigurator() = default;
+	virtual ~AbstractTaskConfigurator() = default;
+
+	virtual std::unique_ptr<AbstractNodecontainer> make() const = 0;
+
+	virtual InputConnector& input(const std::string& name) = 0;
+	virtual InputConnector& input(size_t id) = 0;
```

Comment:
This should be a `uint64_t` according to our discussion yesterday. Also, as we are exposing this to the user, it should be `typedef`ed (or `using ...`) to something like `InputID` so this actually has a semantic meaning.

Reply:
we might want to introduce a using like using Identifier = uint64_t